packages/file_selector/file_selector_platform_interface/lib/src/method_channel/method_channel_file_selector.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import '../platform_interface/file_selector_interface.dart';
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+
+const MethodChannel _channel = MethodChannel('plugins.flutter.io/file_picker');
+
+/// An implementation of [FileSelectorPlatform] that uses method channels.
+class MethodChannelFileSelector extends FileSelectorPlatform {
+  /// Load a file from user's computer and return it as an XFile
+  @override
+  Future<XFile> loadFile({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) async {
+    String path = await _channel.invokeMethod<String>(
+      'loadFile',
+      <String, Object>{
+        'acceptedTypes': acceptedTypeGroups,
+        'initialDirectory': initialDirectory,
+      },
```

Comment:
Do you think this will require a completely separate invokeMethod call to the native side? I think this would be something like:

```dart
_channel.invokeMethod<String>('loadFiles',
  <String, Object>{
        'acceptedTypes': acceptedTypeGroups,
        'initialDirectory': initialDirectory,
        'multiple': false, // boom
  },
```

We can have more methods in the API than in the native side, especially if we're adding them as a "convenience" for users (like in this case! :P)

Reply:
Good catch :) This is actually exactly what I did in the web implementation too!

Reply:
to be clear, this can then be updated in the platform interface too?

In file_selector:
`
loadFile(...) {
  return FileSelectorPlatform.instance.loadFile(... multiple: false, ...)
}
loadFiles(...) {
  return FileSelectorPlatform.instance.loadFile(... multiple: true, ...)
}
`

Reply:
No, I think that if we have two separate methods in the plugin, we can have them in the interface too (they might share private code, though)